app/Services/CrudService.php

```diff
@@ -56,6 +56,16 @@ public function delete(int $id): object
         return $this->repository->delete($entity);
     }
 
+    public function forceDelete(int $id): object
+    {
+        $entity = $this->repository->findTrash($id);
+        if (!$entity) {
+            $errorMessage = call_user_func($this->generateErrorMessage);
+            throw new NotFoundHttpException($errorMessage);
+        }
+        return $this->repository->forceDelete($entity);
+    }
+
```

Comment:
We don't need this method. I've managed CrudService more better way with delete() method.

Considering scenario - 
1. Soft delete
2. Hard Delete
3. Permanently Delete

Reply:
Merge main branch in your local and you would get.

Reply:
Vai tahole ki laravel softDelete bad diye dibo?